### PR TITLE
fix(inventory): update network port metrics

### PR DESCRIPTION
**Pull request by stonebuzz**
Actually ```NetworkPortMetric``` are never update / added after first import of ```NetworkPort```

This PR fix this

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #12483

### CODE REVIEW

**Review by g-bougard — APPROVED** [review]

**Comment by stonebuzz** [comment]
refactor code: now it uses ```portUpdated``` from ```InventoryNetworkPort``` trait

**Review by trasher — APPROVED** [review]

**Review by g-bougard — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

File: src/Inventory/Asset/NetworkPort.php [hunk 1]
```diff
@@ -443,6 +444,26 @@ private function handleVlans(\stdClass $port, int $netports_id)
...
+    private function handleMetrics(\stdClass $port, int $netports_id)
+    {
+        $input = (array)$port;
+        //only update networkport metric if needed
+        if (
+            isset($input['ifinbytes'])
+            && isset($input['ifoutbytes'])
+            && isset($input['ifinerrors'])
+            && isset($input['ifouterrors'])
+        ) {
```
Comment (cedric-anne) [inline src/Inventory/Asset/NetworkPort.php:456]: You can use a shorter notation here.
```suggestion
        if (isset($input['ifinbytes'], $input['ifoutbytes'], $input['ifinerrors'], $input['ifouterrors'])) {
```

File: tests/functionnal/Glpi/Inventory/Assets/NetworkPort.php [hunk 2]
```diff
@@ -352,4 +355,234 @@ public function testPrepare($xml, $ports, $connections, $vlans, $aggregates)
...
+
+        $networkquipement_id = $inventory->getItem()->fields['id'];
+        $this->integer($networkquipement_id)->isGreaterThan(0);
+
+        //get networkEquipement
+        $this->boolean($networkequipment->getFromDB($networkquipement_id))->isTrue();
+
+        //get networkport
+        $this->boolean($networkport->getFromDbByCrit(['itemtype' => 'NetworkEquipment', 'items_id' => $networkquipement_id, 'instantiation_type' => 'NetworkPortEthernet']))
+        ->isTrue();
```
Comment (cedric-anne) [inline tests/functionnal/Glpi/Inventory/Assets/NetworkPort.php:511]: This part seems useless on update.
```suggestion
```

File: tests/functionnal/Glpi/Inventory/Assets/NetworkPort.php [hunk 3]
```diff
@@ -352,4 +355,234 @@ public function testPrepare($xml, $ports, $connections, $vlans, $aggregates)
...
```
Comment (cedric-anne) [inline tests/functionnal/Glpi/Inventory/Assets/NetworkPort.php:561]:
```suggestion
```

File: tests/functionnal/Glpi/Inventory/Assets/NetworkPort.php [hunk 4]
```diff
@@ -352,4 +355,234 @@ public function testPrepare($xml, $ports, $connections, $vlans, $aggregates)
...
+
+        //now we have two metrics, one for yesterday and one for today
```
Comment (cedric-anne) [inline tests/functionnal/Glpi/Inventory/Assets/NetworkPort.php:563]:
```suggestion
        //we still have two metrics, but today metrics are updated
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]
You should also test behaviour for `ifinerrors` and `ifouterrors` fields.

File: tests/functionnal/Glpi/Inventory/Assets/NetworkPort.php (hunk 4 above)
Comment (stonebuzz) [reply tests/functionnal/Glpi/Inventory/Assets/NetworkPort.php:563]: done

File: tests/functionnal/Glpi/Inventory/Assets/NetworkPort.php (hunk 2 above)
Comment (stonebuzz) [reply tests/functionnal/Glpi/Inventory/Assets/NetworkPort.php]: done

File: tests/functionnal/Glpi/Inventory/Assets/NetworkPort.php (hunk 3 above)
Comment (stonebuzz) [reply tests/functionnal/Glpi/Inventory/Assets/NetworkPort.php]: done

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**